Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -3054,4 +3054,9 @@ UserSelect RenderStyle::effectiveUserSelect() const
     return value;
 }
 
+void RenderStyle::setListStyleType(ListStyle listStyle)
+{
+    m_inheritedFlags.listStyleType = static_cast<unsigned>(listStyle.type);
+    setListStyleStringValue(listStyle.name);
+}
```

Comment:
This is rather awkward. You should probably move the whole ListStyleType value to RareInheritedData instead of having the bit in RenderStyle and the text value in rare data. I'd think it is reasonably rare.